File: migrate.py
#!/usr/bin/env python3
import os
import sys
import psycopg2
from psycopg2 import sql

# Database connection details
DB_HOST = "db.enukwrgrbbglxvllcjjn.supabase.co"
DB_PORT = 5432
DB_NAME = "postgres"
DB_USER = "postgres"
DB_PASSWORD = os.environ.get('DB_PASSWORD', '')

if not DB_PASSWORD:
    print("❌ Please set DB_PASSWORD environment variable")
    print("Run: export DB_PASSWORD='your-database-password'")
    print("Get password from: https://supabase.com/dashboard/project/enukwrgrbbglxvllcjjn/settings/database")
    sys.exit(1)

def run_migrations():
    try:
        # Connect to database
        print("🔌 Connecting to database...")
        conn = psycopg2.connect(
            host=DB_HOST,
            port=DB_PORT,
            database=DB_NAME,
            user=DB_USER,
            password=DB_PASSWORD
        )
        conn.autocommit = True
        cursor = conn.cursor()
        print("✅ Connected to database\n")

        # Migration 1: Groups table
        print("📦 Migrating groups table...")
        migrations = [
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS city TEXT DEFAULT 'Unknown City'",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS teacher_title TEXT DEFAULT 'Professor'",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS teacher_full_name TEXT DEFAULT 'Unknown Teacher'",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS capoeira_style TEXT DEFAULT 'contemporanea'",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS lineage TEXT",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS header_image_url TEXT",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS teacher_profile_picture TEXT",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS admin_ids TEXT[] DEFAULT '{}'",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS teacher_ids TEXT[] DEFAULT '{}'",
            "ALTER TABLE groups ADD COLUMN IF NOT EXISTS member_ids TEXT[] DEFAULT '{}'",
        ]
        
        for migration in migrations:
            try:
                cursor.execute(migration)
                print(f"  ✅ {migration.split('ADD COLUMN IF NOT EXISTS')[1].split(' ')[1]}")
            except Exception as e:
                print(f"  ⚠️  {e}")

        # Migration 2: Users table
        print("\n📦 Migrating users table...")
        user_migrations = [
            "ALTER TABLE users ADD COLUMN IF NOT EXISTS teaching_group_ids TEXT[] DEFAULT '{}'",
            "ALTER TABLE users ADD COLUMN IF NOT EXISTS joined_group_ids TEXT[] DEFAULT '{}'",
            "ALTER TABLE users ADD COLUMN IF NOT EXISTS affiliation_group_id TEXT",
            "ALTER TABLE users ADD COLUMN IF NOT EXISTS group_id TEXT",
            "ALTER TABLE users ADD COLUMN IF NOT EXISTS group_name TEXT",
            "ALTER TABLE users ADD COLUMN IF NOT EXISTS teacher_name TEXT",
        ]
        
        for migration in user_migrations:
            try:
                cursor.execute(migration)
                field = migration.split('ADD COLUMN IF NOT EXISTS')[1].split(' ')[1]
                print(f"  ✅ {field}")
            except Exception as e:
                print(f"  ⚠️  {e}")

        # Migration 3: Storage buckets
        print("\n📦 Creating storage buckets...")
        bucket_sql = """
            INSERT INTO storage.buckets (id, name, public)
            VALUES (%s, %s, %s)
            ON CONFLICT (id) DO NOTHING
        """
        buckets = [
            ('profile-pictures', 'profile-pictures', True),
            ('group-images', 'group-images', True),
            ('announcement-images', 'announcement-images', True),
        ]
        
        for bucket in buckets:
            try:
                cursor.execute(bucket_sql, bucket)
                print(f"  ✅ {bucket[0]}")
            except Exception as e:
                print(f"  ⚠️  {bucket[0]}: {e}")

        print("\n🎉 Migrations complete!")
        
        cursor.close()
        conn.close()
        
    except psycopg2.OperationalError as e:
        print(f"❌ Connection failed: {e}")
        print("\nMake sure:")
        print("1. Your password is correct")
        print("2. Get it from: https://supabase.com/dashboard/project/enukwrgrbbglxvllcjjn/settings/database")
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    run_migrations()

File: migrate.js
const { createClient } = require('@supabase/supabase-js');

// Use service role key for admin access
const SUPABASE_URL = 'https://enukwrgrbbglxvllcjjn.supabase.co';
const SUPABASE_SERVICE_KEY = process.env.SUPABASE_SERVICE_KEY || 'your-service-role-key-here';

// Get service key from: https://supabase.com/dashboard/project/enukwrgrbbglxvllcjjn/settings/api
// Look for "service_role" under Project API keys

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_KEY, {
  auth: {
    autoRefreshToken: false,
    persistSession: false
  }
});

async function runMigrations() {
  console.log('🚀 Starting migrations...\n');

  // Test connection
  try {
    const { data, error } = await supabase.from('groups').select('id').limit(1);
    if (error && error.message.includes('column')) {
      console.log('✅ Connected to database, migrations needed\n');
    } else {
      console.log('✅ Connected to database\n');
    }
  } catch (e) {
    console.error('❌ Failed to connect. Check your service key.');
    console.error('Get it from: https://supabase.com/dashboard/project/enukwrgrbbglxvllcjjn/settings/api');
    process.exit(1);
  }

  // Since we can't run ALTER TABLE directly, we'll check what exists
  // and provide feedback
  
  console.log('Checking groups table...');
  try {
    await supabase.from('groups').select('city, teacher_title').limit(1);
    console.log('✅ Groups table has new columns');
  } catch (e) {
    console.log('❌ Groups table missing columns');
    console.log('   Run the SQL from supabase/migrations/20250818184535_add_enhanced_group_fields.sql');
  }

  console.log('\nChecking users table...');
  try {
    await supabase.from('users').select('teaching_group_ids').limit(1);
    console.log('✅ Users table has new columns');
  } catch (e) {
    console.log('❌ Users table missing columns');
    console.log('   Run the SQL from supabase/migrations/20250818184535_add_enhanced_group_fields.sql');
  }

  console.log('\nChecking storage buckets...');
  try {
    const buckets = await supabase.storage.listBuckets();
    const required = ['profile-pictures', 'group-images', 'announcement-images'];
    
    for (const bucket of required) {
      if (buckets.data?.some(b => b.id === bucket)) {
        console.log(`✅ Bucket exists: ${bucket}`);
      } else {
        console.log(`❌ Missing bucket: ${bucket}`);
        // Try to create it
        try {
          await supabase.storage.createBucket(bucket, { public: true });
          console.log(`   ✅ Created bucket: ${bucket}`);
        } catch (e) {
          console.log(`   ❌ Could not create bucket: ${e.message}`);
        }
      }
    }
  } catch (e) {
    console.log('❌ Could not check storage buckets');
  }

  console.log('\n📝 Migration check complete!');
  console.log('If you see ❌ above, you need to run the migrations.');
  console.log('\nTo get your service key:');
  console.log('1. Go to: https://supabase.com/dashboard/project/enukwrgrbbglxvllcjjn/settings/api');
  console.log('2. Copy the "service_role" key (starts with eyJ...)');
  console.log('3. Run: SUPABASE_SERVICE_KEY="your-key" node migrate.js');
}

// Check if service key is provided
if (SUPABASE_SERVICE_KEY === 'your-service-role-key-here') {
  console.error('❌ Please provide your Supabase service role key!');
  console.error('\nGet it from: https://supabase.com/dashboard/project/enukwrgrbbglxvllcjjn/settings/api');
  console.error('Look for "service_role" under Project API keys\n');
  console.error('Then run: SUPABASE_SERVICE_KEY="your-key" node migrate.js');
  process.exit(1);
}

runMigrations().catch(console.error);